shared/src/vyxal/VAny.scala

```diff
@@ -126,7 +128,17 @@ extension (self: VAny)
       case s: String => s.nonEmpty
       case f: VFun => true
       case l: VList => l.nonEmpty
+      case c: VConstructor => true
+      case o: VObject => true
 end extension
 
+case class VConstructor(
+    name: String
+)
+
+case class VObject(
+    className: String,
+    fields: mut.Map[String, (Visibility, VAny)],
```

Comment:
Given that setting a field always makes a new copy of the object, it'd probably be better to make this an immutable map instead